### ISSUE TITLE
JSR-352 Job Operator - Can Neither Abandon nor Stop job post crash [BATCH-2433]

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-batch#1169]
**[Brian Rogers](https://jira.spring.io/secure/ViewProfile.jspa?name=xerofun)** opened **[BATCH-2433](https://jira.spring.io/browse/BATCH-2433?redirect=false)** and commented

The logic for abandon or stop does not cover all failure cases.  My guess is that because stop uses JSR Batch status and abandon uses Spring's JobExecution.isRunning method.

When calling abandon I receive:

```java
javax.batch.operations.JobExecutionIsRunningException: Unable to abandon a job that is currently running
```

When calling stop on the same job I get the seemingly contrary message:

```java
javax.batch.operations.JobExecutionNotRunningException: JobExecution must be running so that it can be stopped: JobExecution: id=448, version=2, startTime=2015-09-09 11:26:13.543, endTime=null, lastUpdated=2015-09-09 14:26:42.003, status=STOPPING, exitStatus=exitCode=UNKNOWN;exitDescription=; <redacted>
```

I'm having a difficult time reproducing the "STOPPING" state though I imagine the JVM was shut down during the termination of my batch job which has a long running partitioned chunk step.  I also have some other jobs in the "STARTED" state that are clearly not running as the JVM has been recycled multiple times since.

Basically there needs to be a reliable way of cleaning up these post-crash jobs through the JSR JobOperator interface.  It may be as simple as updating the checks in STOP and ABANDON but I don't have a full appreciation of what's going on behind the scenes in all possible scenarios.

The other thing to note is that JobOperator.getRunningExecutions is returning these "STOPPING" instances.  Which is probably not incorrect behavior, just another data point.

---

**Affects:** 3.0.1, 3.0.2, 3.0.3, 3.0.4, 3.0.5

**Reference URL:** https://gist.github.com/xerofun/ac8c19ef814253efedf5#file-gistfile1-java

**Comment by fmbenhassine** [comment]
The implementation of the JSR-352 was [discontinued in v5](https://github.com/spring-projects/spring-batch/wiki/Spring-Batch-5.0-Migration-Guide#jsr-352-implementation-removal).

If anyone is still having this issue with v4.3.8+, then please provide a [minimal complete verifiable example](https://github.com/spring-projects/spring-batch/blob/main/ISSUE_REPORTING.md)  that we can run to reproduce the problem and we will re-open this issue for further investigation. Thank you upfront!
